public/javascripts/main.js

```diff
@@ -1,72 +1,129 @@
+        } else {
+          row.classList.add("invisible");
+          row.querySelector(".search-cell").innerHTML = "";
+        }
+      }
+    });
+  } else {
+    Array.from(document.querySelectorAll(".searchable-row")).forEach(function (row) { row.classList.remove("invisible"); });
+    Array.from(document.querySelectorAll(".search-cell")).forEach(function (cell) { cell.innerHTML=""; });
+    Array.from(document.querySelectorAll("tfoot")).forEach(function (row) { row.classList.add("invisible") });
+  }
 }
 
 function clearSearch() {
-    document.getElementById("search-input").value = "";
-    onSearch();
+  document.getElementById("search-input").value = "";
+  onSearch();
 }
 
 function changeMDLInputChecked(input, isChecked) {
-    input.checked = isChecked;
-    if(isChecked) {
-        input.parentNode.classList.add("is-checked");
-    } else {
-        input.parentNode.classList.remove("is-checked");
-    }
-    componentHandler.upgradeElements(input);
+  input.checked = isChecked;
+  if(isChecked) {
+    input.parentNode.classList.add("is-checked");
+  } else {
+    input.parentNode.classList.remove("is-checked");
+  }
+  componentHandler.upgradeElements(input);
 }
 
 
 function deleteElement(selector) {
-    var element = document.querySelector(selector);
-    element.parentNode.removeChild(element);
+  var element = document.querySelector(selector);
+  element.parentNode.removeChild(element);
 }
 
 function checkAllBySelector(selector) {
-    var checkboxes = window.document.querySelectorAll(selector);
-    for (var i = 0; i < checkboxes.length; i++) {
-        checkboxes[i].checked = true;
-        checkboxes[i].parentElement.classList.add("is-checked");
-    }
+  var checkboxes = window.document.querySelectorAll(selector);
+  for (var i = 0; i < checkboxes.length; i++) {
+    checkboxes[i].checked = true;
+    checkboxes[i].parentElement.classList.add("is-checked");
+  }
 }
 
 function uncheckAllBySelector(selector) {
-    var checkboxes = window.document.querySelectorAll(selector);
-    for (var i = 0; i < checkboxes.length; i++) {
-        checkboxes[i].checked = false;
-        checkboxes[i].parentElement.classList.remove("is-checked");
-    }
+  var checkboxes = window.document.querySelectorAll(selector);
+  for (var i = 0; i < checkboxes.length; i++) {
+    checkboxes[i].checked = false;
+    checkboxes[i].parentElement.classList.remove("is-checked");
+  }
 }
 
 
 var Main = {
-    disableEventTarget: function(element) {
-        element.disabled = 'true';
-    }
-}
\ No newline at end of file
+  disableEventTarget: function(element) {
+    element.disabled = 'true';
+  }
+}
+
+//
+// Form Functions
+//
+
+var aplusFormHasBeenSubmitted = false;
+
+// https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload
+function setupOnbeforeunload(event) {
+  var message = "En quittant la page, les données ne seront pas enregistrées.";
```

Comment:
```suggestion
  var message = "Merci de confirmer la fermeture de cette page, les données saisies ne seront pas enregistrées.";
```
On peut le faire mais je crois que ça n'a pas d'impact sous chrome et firefox.